mail_tracking_mailgun/tests/test_mailgun.py

```diff
@@ -0,0 +1,460 @@
+                },
+                "severity": "permanent",
+            }
+        )
+        with self._request_mock():
+            self.MailTrackingController.mail_tracking_mailgun_webhook()
+        event = self.event_search("hard_bounce")
+        self.assertEqual(event.timestamp, float(self.timestamp))
+        self.assertEqual(event.recipient, self.recipient)
+        self.assertEqual(event.error_type, str(code))
+        self.assertEqual(event.error_description, error)
+        self.assertEqual(event.error_details, notification)
+
+    def test_event_rejected(self):
+        reason = "hardfail"
+        description = "Not delivering to previously bounced address"
+        self.event.update(
+            {
+                "event": "rejected",
+                "reject": {"reason": reason, "description": description},
+            }
+        )
+        with self._request_mock():
+            self.MailTrackingController.mail_tracking_mailgun_webhook()
+        event = self.event_search("reject")
+        self.assertEqual(event.timestamp, float(self.timestamp))
+        self.assertEqual(event.recipient, self.recipient)
+        self.assertEqual(event.error_type, "rejected")
+        self.assertEqual(event.error_description, reason)
+        self.assertEqual(event.error_details, description)
+
+    @mock.patch(_packagepath + ".models.res_partner.requests")
+    def test_email_validity(self, mock_request):
+        self.partner.email_bounced = False
+        mock_request.get.return_value.apparent_encoding = "ascii"
+        mock_request.get.return_value.status_code = 200
+        mock_request.get.return_value.json.return_value = {
+            "is_valid": True,
+            "mailbox_verification": "true",
+        }
+        # Trigger email auto validation in partner
+        self.env["ir.config_parameter"].set_param(
+            "mailgun.auto_check_partner_email", "True"
+        )
+        self.partner.email = "info@tecnativa.com"
+        self.assertFalse(self.partner.email_bounced)
+        self.partner.email = "xoxoxoxo@tecnativa.com"
+        # Not a valid mailbox
+        mock_request.get.return_value.json.return_value = {
+            "is_valid": True,
+            "mailbox_verification": "false",
+        }
+        with self.assertRaises(UserError):
+            self.partner.check_email_validity()
+        # Not a valid mail address
+        mock_request.get.return_value.json.return_value = {
+            "is_valid": False,
+            "mailbox_verification": "false",
+        }
+        with self.assertRaises(UserError):
+            self.partner.check_email_validity()
+        # Unable to fully validate
+        mock_request.get.return_value.json.return_value = {
+            "is_valid": True,
+            "mailbox_verification": "unknown",
+        }
+        with self.assertRaises(UserError):
+            self.partner.check_email_validity()
+        self.assertTrue(self.partner.email_bounced)
```

Comment:
Here's the test fix you need

```suggestion
        with self.assertRaises(UserError):
            self.partner.check_email_validity()
        # If we autocheck, the mail will be bounced
        self.partner.with_context(mailgun_auto_check=True).check_email_validity()
        self.assertTrue(self.partner.email_bounced)
        # Unable to fully validate
        mock_request.get.return_value.json.return_value = {
            "is_valid": True,
            "mailbox_verification": "unknown",
        }
        with self.assertRaises(UserError):
            self.partner.check_email_validity()
```